R-TypeGame/Systems/ShootMob/ShootSystem.cpp

```diff
@@ -24,33 +27,56 @@ void ShootSystem::_shootIfNeeded(ECS::Registry& reg, int idxPacketEntities)
-                ShootInitSystem().getFunction()(reg, shoot);
-                positions = reg.get_components<IComponent>("Position2DComponent");
-                std::shared_ptr<Position2DComponent> positionShoot = std::dynamic_pointer_cast<Position2DComponent>(positions[shoot]);
-                if (!positionShoot)
-                    return;
-                ECS::SparseArray<IComponent> scales = reg.get_components<IComponent>("ScaleComponent");
-                if (scales.size() <= entity)
-                    return;
-                std::shared_ptr<ScaleComponent> scale = std::dynamic_pointer_cast<ScaleComponent>(scales[entity]);
-                ECS::SparseArray<IComponent> texturesRect = reg.get_components<IComponent>("TextureRectComponent");
-                std::shared_ptr<TextureRectComponent> textureRect = std::dynamic_pointer_cast<TextureRectComponent>(texturesRect[entity]);
-                positionShoot->x = positionMob->x + ((textureRect->width / 2) * scale->scale);
-                positionShoot->y = positionMob->y + ((textureRect->height / 2) * scale->scale);
+                if (!positionMob)
+                    continue;
+                ShootTypeComponent::ShootType shootType = std::dynamic_pointer_cast<ShootTypeComponent>(shootTypes[entity])->type;
+                if (shootType == ShootTypeComponent::ShootType::NONE)
+                    continue;
+                std::string mobSpritePath = std::dynamic_pointer_cast<TextureRectComponent>(TextRec[entity])->path;
+                for (int i = 0; i < shootType + 1; i++) {
+                    ECS::entity_t shoot = reg.spawn_entity();
+                    if (mobSpritePath == "./Assets/mob-spe-shot-1.gif") {
+                        MobSpeOneShootInitSystem().getFunction()(reg, shoot);
+                    } else if (mobSpritePath == "./Assets/shots.gif") {
+                        ShootInitSystem().getFunction()(reg, shoot);
+                    } else {
+                        MobSpeTwoShootInitSystem().getFunction()(reg, shoot);
+                    }
+                    positions = reg.get_components<IComponent>("Position2DComponent");
+                    std::shared_ptr <Position2DComponent> positionShoot = std::dynamic_pointer_cast<Position2DComponent>(positions[shoot]);
+                    if (!positionShoot)
+                        return;
+                    ECS::SparseArray<IComponent> scales = reg.get_components<IComponent>("ScaleComponent");
+                    if (scales.size() <= entity)
+                        return;
+                    std::shared_ptr <ScaleComponent> scale = std::dynamic_pointer_cast<ScaleComponent>(scales[entity]);
+                    ECS::SparseArray<IComponent> texturesRect = reg.get_components<IComponent>("TextureRectComponent");
+                    std::shared_ptr <TextureRectComponent> textureRect = std::dynamic_pointer_cast<TextureRectComponent>(texturesRect[entity]);
```

Comment:
Same issue for the out of range index

Reply:
Request me and not Théophile, I probably will cry soon:(

Reply:
I maybe missed it or wasn't a reviewer for the pr